<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MagicEraserWidget</class>
 <widget class="QWidget" name="MagicEraserWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>200</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>6</number>
   </property>
   <property name="leftMargin">
    <number>9</number>
   </property>
   <property name="topMargin">
    <number>9</number>
   </property>
   <property name="rightMargin">
    <number>9</number>
   </property>
   <property name="bottomMargin">
    <number>9</number>
   </property>
   <item>
    <widget class="QCheckBox" name="active_checkbox">
     <property name="toolTip">
      <string>Enable/disable magic eraser tool</string>
     </property>
     <property name="text">
      <string>Active</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="brush_size_layout">
     <item>
      <widget class="QLabel" name="brush_size_label">
       <property name="text">
        <string>Brush Size:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="brush_size_spinbox">
       <property name="toolTip">
        <string>Size of the eraser brush (1-100 pixels)</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="filter_size_layout">
     <item>
      <widget class="QLabel" name="filter_size_label">
       <property name="text">
        <string>Median Filter Size:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="filter_size_spinbox">
       <property name="toolTip">
        <string>Size of the median filter kernel (3-101, must be odd)</string>
       </property>
       <property name="minimum">
        <number>3</number>
       </property>
       <property name="maximum">
        <number>101</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>25</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPushButton" name="drawing_mode_button">
     <property name="toolTip">
      <string>Toggle drawing mode to paint areas for erasing</string>
     </property>
     <property name="text">
      <string>Start Drawing</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="clear_mask_button">
     <property name="toolTip">
      <string>Clear the current magic eraser mask</string>
     </property>
     <property name="text">
      <string>Clear Mask</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="instructions_label">
     <property name="text">
      <string>Instructions: Enable drawing mode and paint over areas to erase. 
The erased areas will be replaced with median-filtered content.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui> 